task07-consume-real-data/data.js

```diff
@@ -0,0 +1,75 @@
+//import fetch from 'node-fetch';
+
+var characters = [];
+var islands = [];
+var mysticObjects = [];
+
+const sortByName = function(a, b){
+  let x = a.name.toLowerCase();
+  let y = b.name.toLowerCase();
+  if (x < y) {return -1;}
+  if (x > y) {return 1;}
+  return 0;
+};
+
+function fetchData() {
+  fetch('https://static.akurey.com/trainings/OnePieceInformation.json')
+  .then( response => response.json())
+  .then( data => {
+    characters = data.characters;
+    islands = data.islands;
+    mysticObjects = data.mysticObjects;
+  })
+  .then( () => {
+    characters.sort(sortByName);
+    islands.sort(sortByName);
+    mysticObjects.sort(sortByName);
+    createCharacters(characters.slice(0,4));
+    createIslands(islands.slice(0,3));
+    createobjects(mysticObjects.slice(0,4));
+  })
```

Comment:
Thanks for your feedback! you are totally right, the last .then() is unnecessary but I still need the second one since response.json() returns a promise